### ISSUE TITLE
use MTK macro processing code instead of writing custom DSL code

### ISSUE THREAD

**Issue by isaacsas** [SciML/Catalyst.jl#1023]
On a quick skim, it appears large subsets of our DSL macro processing code overlap with MTK macro processing code. We should switch to MTK's code as much as possible to avoid needing to maintain our own macros (for example `@equation`), and to ensure feature compatibility with MTK.

**Comment by isaacsas** [comment]
OK, good. I missed that we use `parse_equations!` now (there is a lot of DSL code at this point...).

**Comment by TorkelE** [comment]
yeah, the DSL code has gotten increasingly messy over the last while. I tried to clean it up a bit in https://github.com/SciML/Catalyst.jl/pull/985, but even after that more could probably be done to get it up to the standard of the rest of the package.
